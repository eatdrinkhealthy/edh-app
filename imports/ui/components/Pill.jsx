// @flow
import React from "react";
import classNames from "classnames";

type IPillProps = {
  className?: string, // eslint-disable-line react/require-default-props
  name: string,
  active: boolean,
  selectedColor?: string,
  onClick: (*) => void,
};

const Pill = ({
  className: excludeClassName, // extract out and override className
  active, // use active, but don't pass it on to button
  selectedColor,
  ...props // pass on the rest of the props (used by a button)
}: IPillProps) => {
  // TODO find better way to optimize styling (all here, or in .less file)
  const colorStyle = {
    background: active ? selectedColor : "white",
    borderColor: active ? selectedColor : "#ccc",
  };

  return (
    <button
      type="button"
      style={colorStyle}
      className={classNames("pill", { pill_selected: active })}
      {...props}
    />
  );
};

Pill.defaultProps = {
  selectedColor: "#5e2ca5",
};

export default Pill;
