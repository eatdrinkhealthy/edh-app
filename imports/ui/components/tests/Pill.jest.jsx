// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Pill from "../Pill";

describe("<Pill />", function () {
  it("matches render snapshot - active", function () {
    const tree = renderer.create(
      <Pill name="pill1" onClick={() => {}} active>
        Test Pill Button
      </Pill>,
    );
    expect(tree).toMatchSnapshot();
  });

  it("matches render snapshot - inactive", function () {
    const tree = renderer.create(
      <Pill name="pill1" onClick={() => {}} active={false}>
        Test Pill Button
      </Pill>,
    );
    expect(tree).toMatchSnapshot();
  });

  it("should not pass a className prop on to the button component", function () {
    const wrapper = shallow(
      <Pill name="pill1" onClick={() => {}} active={false} className="badClass">
        Test Pill Button
      </Pill>,
    );
    const button = wrapper.find("button");
    expect(button.prop("className")).toBe("pill");
    expect(button.prop("name")).toBe("pill1");
    expect(button.prop("active")).toBe(false);
  });
});
