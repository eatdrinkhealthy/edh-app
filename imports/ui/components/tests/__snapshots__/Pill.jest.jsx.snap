// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill /> matches render snapshot - active 1`] = `
<button
  className="pill pill_selected"
  name="pill1"
  onClick={[Function]}
>
  Test Pill Button
</button>
`;

exports[`<Pill /> matches render snapshot - inactive 1`] = `
<button
  className="pill"
  name="pill1"
  onClick={[Function]}
>
  Test Pill Button
</button>
`;
